### PR TITLE
Add Karabiner Elements v0.90.46

### LINKED ISSUE #24895: Cask request: Karabiner Elements
### Cask details

(Please fill out as much as possible)

**Name** -
Karabiner-Elements

**Homepage** -
https://github.com/tekezo/Karabiner-Elements

**License** -
Unlicense

**Download URL** -
https://pqrs.org/latest/karabiner-elements-latest.dmg

**Description** -
The replacement for Seil (https://github.com/caskroom/homebrew-cask/blob/master/Casks/seil.rb) for macOS Sierra. And appearantly somewhat for Karabiner as well
https://pqrs.org/osx/karabiner/seil.html.en
image

### PR DESCRIPTION
**Pull request by johndbritton**
After making all changes to the cask:
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Aditionally, **if adding a new cask**:
- [x] Named the cask according to the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [ ] Checked that the cask was not already refused in [closed issues](https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed).
  - https://github.com/caskroom/homebrew-cask/pull/24988#pullrequestreview-2125317 exists, but hasn't had recent progress. Didn't see this until after I did all the work to make this cask.

Closes https://github.com/caskroom/homebrew-cask/issues/24895.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
This cask is less complete than the current PR. For instance, it’s missing `uninstall` (mandatory for `pkg` casks). If you make the fixes requested in the other PR, we’ll merge yours instead.

**Comment by johndbritton** [comment]
>  If you make the additions and fixes requested in the other PR, we’ll merge yours instead.

Updated to include appcast and uninstall stanzas as requested in https://github.com/caskroom/homebrew-cask/pull/24988#pullrequestreview-2125317

File: Casks/karabiner-elements.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  name 'Karabiner Elements'
+  homepage 'https://pqrs.org/osx/karabiner/'
+  license :public_domain
+
+  auto_updates true
+
+  pkg 'Karabiner-Elements.sparkle_guided.pkg'
+
+  uninstall pkgutil: 'org.pqrs.Karabiner-Elements',
+            script:  '/Library/Application Support/org.pqrs/Karabiner-Elements/uninstall.sh'
```
Comment (vitorgalvao) [inline Casks/karabiner-elements.rb]: See [`xampp`](https://github.com/caskroom/homebrew-cask/blob/master/Casks/xampp.rb) as an example.

Add `sudo: false` and see if it fails (it likely will). If it does, make it `sudo: true`. We’re [trying to make all of these explicit](https://github.com/caskroom/homebrew-cask/issues/13608).

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by johndbritton** [comment]
`sudo: false` caused permissions errors so I added `sudo: true`.

**Review by vitorgalvao — APPROVED** [review]

**Comment by vitorgalvao** [comment]
Thank you. Merging after Travis passes.

**Pull request merged**